### DISCUSSION TITLE
Multiple basePath entries for a single service

### DISCUSSION THREAD

**Discussion by madsgodvinjensen** [vercel/next.js#27244] [Ideas]
### Describe the feature you'd like to request

We are preparing to use Next.js to build a micro service/micro frontend architecture.

The plan is to host multiple independant ECS containers in AWS, all served by the same CloudFront distribution. There will be one domain name and each independant service will host one or more paths under that domain name.

The `basePath`/zones feature of Next.js is built just for that. We do see one issue that I'l like to enquire about and offer a potential solution for as well.

The `basePath` is a single value which encourages the architecture to be divided by area of the website rather than by intended user flow for example. Either that, or change URLs to artificially pull things together that would benefit from being in the same service.

We generally think of splitting services by user flow/funnel rather than organisational structure or URL. The benefits for the user will be that we can offer Next.js client side routing plus have a lot of JS chunks cached on the first request and the next in the funnel will be much much faster.

### Describe the solution you'd like

Currently, we'd configure the basePath like this:

```
module.exports = {
  basePath: "/blog",
}
```

Would it be possible to specify this as an array of paths to listen for?

```
module.exports = {
  basePath: ["/blog", "/about],
}
```

This would make it very flexible to structure the application by whatever need we might have rather than by how the URL scheme should be constructed, how our team structure is or similar that the users shouldn't have to pay for.

### Describe alternatives you've considered

The two workarounds we're considering are:

1. Change our URL scheme to fit the user funnel
This is not great because it will have a large impact on the scope of the project. The URLs make sense as they are and changing them will impact SEO, links, bookmarks, support articles etc.

2. Create a lot more services than needed.
This impacts the cacheability as mentioned above, removes the option to use client side linking, and users will pay for it.

**Comment by klonwar** [comment]
+1 on this

Our use case: we want to have one app instance but several localized base paths for different markets (EN, KZ, AZ, ...)
So, for example, we need the app to handle base paths like `site.com/app`, `site.kz/qosymsha`, `site.az/tetbiq`
